File: bank-app/app/src/main/java/com/android/bankapp/login/LoginRouter.java
package com.android.bankapp.login;

import android.content.Context;

import com.android.bankapp.statements.StatementsActivity_;

import java.lang.ref.WeakReference;

public class LoginRouter implements LoginRouterInput {
    public WeakReference<LoginActivity> activity;

    @Override
    public void goToNextScreen(Context context) {
        StatementsActivity_.intent(context).start();
    }
}

File: bankapp/app/src/main/java/dev/ornelas/bankapp/ui/login/LoginRouter.kt
package dev.ornelas.bankapp.ui.login

import android.content.Intent
import dev.ornelas.bankapp.ui.statements.StatementsActivity
import java.lang.ref.WeakReference

class LoginRouter(val activity: WeakReference<LoginActivity>) : LoginContract.Router {

    companion object {
        const val APP_USER_INTENT = "APP_USER_INTENT"
    }

    override fun navigateToStatements(user: UserViewModel) {
        val intent = Intent(activity.get(), StatementsActivity::class.java)
        intent.putExtra(APP_USER_INTENT, user)

        activity.get()?.let {
            it.startActivity(intent)
            it.finish()
        }
    }
}